deepinv/training/trainer.py

```diff
@@ -59,85 +56,150 @@ class Trainer:
+    :param str device: Device on which to run the training (e.g., 'cuda' or 'cpu'). Default is 'cuda' if available, otherwise 'cpu'.
+
+    |sep|
+
+    :Optimization:
+
+    :param None, torch.optim.Optimizer optimizer: Torch optimizer for training the network. Default is ``None``.
+    :param int epochs: Number of training epochs.
+        Default is 100. The trainer will perform gradient steps equal to the `min(epochs*n_batches, max_batch_steps)`.
+    :param int max_batch_steps: Number of gradient steps per iteration.
+        Default is 1e10. The trainer will perform gradient steps equal to the `min(epochs*n_batches, max_batch_steps)`.
+    :param None, torch.optim.lr_scheduler.LRScheduler scheduler: Torch scheduler for changing the learning rate across iterations. Default is ``None``.
+    :param bool early_stop: If ``True``, the training stops when the evaluation loss is not improving. Default is ``False``.
+        The user can modify the strategy for saving the best model by overriding the :func:`deepinv.Trainer.stop_criterium` method.
     :param deepinv.loss.Loss, list[deepinv.loss.Loss] losses: Loss or list of losses used for training the model.
         Optionally wrap losses using a loss scheduler for more advanced training.
         :ref:`See the libraries' training losses <loss>`. By default, it uses the supervised mean squared error.
-        Where relevant, the underlying metric should have ``reduction=None`` as we perform the averaging using :class:`deepinv.utils.AverageMeter` to deal with uneven batch sizes.
+        Where relevant, the underlying metric should have ``reduction=None`` as we perform the averaging
+        using :class:`deepinv.utils.AverageMeter` to deal with uneven batch sizes. Default is :class:`supervised loss <deepinv.loss.SupLoss>`.
+    :param float grad_clip: Gradient clipping value for the optimizer. If None, no gradient clipping is performed. Default is None.
+
+    |sep|
+
+    :Evaluation:
+
     :param None, torch.utils.data.DataLoader, list[torch.utils.data.DataLoader] eval_dataloader: Evaluation data loader(s)
-        should provide a signal x or a tuple of (x, y) signal/measurement pairs or a tuple `(x, y, params)`
-        where `params` is a dict of physics generator parameters to be loaded into the physics each iteration.
-    :param None, torch.optim.lr_scheduler.LRScheduler scheduler: Torch scheduler for changing the learning rate across iterations.
-    :param bool online_measurements: Generate the measurements in an online manner at each iteration by calling
-        ``physics(x)``. This results in a wider range of measurements if the physics' parameters, such as
-        parameters of the forward operator or noise realizations, can change between each sample;
-        the measurements are loaded from the training dataset.
+        should provide a signal `x` or a tuple of `(x, y)` signal/measurement pairs or a tuple `(x, y, params)`
```

Comment:
Similarly here remove description and refer to list of options above on how to pass dataloaders